### PR TITLE
Remove transitive interfaces (implicitly inherited ones)

### LINKED ISSUE #10: How to remove transitive dependencies? (copy from clue/graph)
This is a copy from https://github.com/clue/graph/issues/85

In my class diagrams I have unnecessary links cluttering the display

See in this one having (in the upper left corner) links to Traversable both from Child class and it's parent.

drush-uml  

How can I remove these? Use an algorithm?

### PR DESCRIPTION
**Pull request by clue**
Before:
uml-old

After:
uml-new

Fixes #10.
Not sure how this relates to #17.

### CODE REVIEW

**Comment by clemens-tolboom** [comment]
The result looks awesome :)

But this needs a test somewhat in line with #17 (in which I failed to write a proper one :()

File: src/Fhaculty/Graph/Uml/ClassDiagramBuilder.php [hunk 1]
```diff
@@ -543,4 +543,27 @@ private function escape($id)
...
+        if ($parent) {
+            foreach ($parent->getInterfaceNames() as $in) {
+                unset($interfaces[$in]);
+            }
+        }
+
+        // remove each interface already implemented by any of the inherited interfaces
+        foreach ($interfaces as $if) {
+            foreach ($if->getInterfaceNames() as $in) {
+                //unset($interfaces[$in]);
```
Comment (clemens-tolboom) [inline src/Fhaculty/Graph/Uml/ClassDiagramBuilder.php]: This is not OK ... this commented unset makes the loops unless.

Should we remove all?

Comment (clue) [inline src/Fhaculty/Graph/Uml/ClassDiagramBuilder.php]: Duh, did I actually just comment out _the one line_ that actually fixes this? :>

I've pushed an update :D

Yeah, this was a last-minute change in order to produce the "old" graph. Looks like we need some tests :)

**Comment by clemens-tolboom** [comment]
Lol ... working both with the speed of light.

**Comment by clue** [comment]
Yeah, what a coincidence :)

Feel free to merge this if it's reviewed.

**Comment by clemens-tolboom** [comment]
Awesome fix :)

**Pull request merged**